### ISSUE TITLE
Renewal of multiple certificates fail if one fails

### ISSUE THREAD

**Issue by joelnotified** [sjkp/letsencrypt-siteextension#307]
Hi,
Thank you for a great addition to the community. This has really helped me a lot.

I might be using the extension "wrong" but I have multiple hostnames bound to the same App Service, all with a certificate of their own. For example for the App Service example.com I have the following hostname bindings setup:
- one.example.com
- two.example.com
- three.example.com

The domains are all setup as CNAME redirects at different domain providers, and they point to the corresponding azure for the App Service.

I have one certificated per domain, so it's not one certificate for all of the domains. This all works as expected and certificates are renewed when needed.

However, should one renewal fail, it seems that all subsequent renewals also fail. For example
1. The certificate for one.example.com is renewed
2. The certificate for two.example.com is not renewed. The function RenewCertificate fails with an exception because the CNAME redirect is broken (or whatever reason)
3. The certificate for three.example.com is not renewed, because the job failed in the previous step.

Would it be possible to keep going after the failure of a renewal? So that at least the domains that are working can get renewed?

**Comment by modemgeek** [comment]
Yes, this has been an issue for us as well.  we have up to 200 certificates on a webapp, and sometimes we don't have control over the DNS.  we will have a customer cancel w/o letting us know.  the one failed renewal will hold up the rest.  I can usually find it by going to the webjobs log and delete it from the renewnal process.  However, it would be nice to get an email when one fails to let us know which one and to also continue renewal for the rest of the certificates.

**Comment by porky9** [comment]
We are having exactly the same issue. If a customer cancels or lets their domain name expire the renewal of that certificate will fail and the job will stop processing the rest of the certificates that are due for renewal. I think it would be better if the error was logged and the job carried on to the next certificate that needs to be renewed. Or at the very least it could send an email to notify you of the failed renewal so that you can delete that hostname from Azure.

**Comment by obrousseau** [comment]
Same situation here.

Best scenario would be to renew all certificates that can be renewed and some sort of alert to inform of the ones who couldn't.

**Comment by vigubik** [comment]
Hi, 
Great extension.

But we also experience same issue.
It would be good to collect all errors in some Aggregate exception without interrupting iteration and throw it only after all certificates got chance to be renewed.
